routing/http/client/transport.go

```diff
@@ -36,3 +43,37 @@ func (l *limitReadCloser) Read(p []byte) (int, error) {
 	}
 	return n, err
 }
+
+// ImportPath is the canonical import path that allows us to identify
+// official client builds vs modified forks, and use that info in User-Agent header.
+var ImportPath = importPath()
+
+// importPath returns the path that library consumers would have in go.mod
+func importPath() string {
+	p := reflect.ValueOf(ResponseBodyLimitedTransport{}).Type().PkgPath()
+	// we have monorepo, so stripping the remainder
+	return strings.TrimSuffix(p, "/routing/http/client")
+}
+
+// moduleVersion returns a useful user agent version string allowing us to
+// identify requests coming from official releases of this module vs forks.
+func moduleVersion() (ua string) {
+	ua = ImportPath
+	var module *debug.Module
+	if bi, ok := debug.ReadBuildInfo(); ok {
+		// If debug.ReadBuildInfo was successful, we can read Version by finding
+		// this client in the dependency list of the app that has it in go.mod
+		for _, dep := range bi.Deps {
+			if dep.Path == ImportPath {
+				module = dep
+				break
+			}
+		}
+		if module != nil {
+			ua += "@" + module.Version
+			return
+		}
+		ua += "@unknown"
+	}
+	return
+}
```

Comment:
This might be a bit complex. Can we just have a default string header as a simpler default? something like `router/http/go`

Reply:
Defaulting to `User-Agent: Go-http-client/1.1` is not bad, but does not tell us when someone used our library, and when they wrote their own client.  

Having dynamic default set to package name will tell us what version of lib is in the use, how fast people upgrade, and if they fork, we will see the repo name and be able to inspect what changes were made.

Overall, less guesswork for IPFS stewards, more visibility into ecosystem.